hive/src/main/java/org/apache/iceberg/hive/legacy/LegacyHiveTableOperations.java

```diff
@@ -0,0 +1,226 @@
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.hive.HiveClientPool;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class LegacyHiveTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LegacyHiveTableOperations.class);
+
+  private final HiveClientPool metaClients;
+  private final String database;
+  private final String tableName;
+  private final Configuration conf;
+
+  private FileIO fileIO;
+
+  protected LegacyHiveTableOperations(Configuration conf, HiveClientPool metaClients, String database, String table) {
+    this.conf = conf;
+    this.metaClients = metaClients;
+    this.database = database;
+    this.tableName = table;
+  }
+
+  @Override
+  public FileIO io() {
+    if (fileIO == null) {
+      fileIO = new HadoopFileIO(conf);
+    }
+
+    return fileIO;
+  }
+
+  @Override
+  protected void doRefresh() {
+    try {
+      org.apache.hadoop.hive.metastore.api.Table hiveTable =
+          metaClients.run(client -> client.getTable(database, tableName));
+
+      Schema schema = LegacyHiveTableUtils.getSchema(hiveTable);
+      PartitionSpec spec = LegacyHiveTableUtils.getPartitionSpec(hiveTable, schema);
+
+      TableMetadata metadata = TableMetadata.newTableMetadata(schema, spec, hiveTable.getSd().getLocation(),
+          LegacyHiveTableUtils.getTableProperties(hiveTable));
+      metadata = metadata.replaceCurrentSnapshot(getDummySnapshot());
+      setCurrentMetadata(metadata);
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
+    }
+    setShouldRefresh(false);
+  }
+
+  DirectoryInfo getDirectoryInfo() {
+    Preconditions.checkArgument(current().spec().fields().isEmpty(),
+        "getDirectoryInfo only allowed for unpartitioned tables");
+    try {
+      org.apache.hadoop.hive.metastore.api.Table hiveTable =
+          metaClients.run(client -> client.getTable(database, tableName));
+
+      return LegacyHiveTableUtils.toDirectoryInfo(hiveTable);
+    } catch (TException e) {
+      String errMsg = String.format("Failed to get table info from metastore %s.%s", database, tableName);
+      throw new RuntimeException(errMsg, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during refresh", e);
```

Comment:
Seems like this method is not being called from refresh at all. Seems strange to have these utilities methods here. I guess you require some existing correctly initialized vars which is provided by metacleints and conf for example.

I'm ok keeping these methods here. Maybe we can move the listFiles method here as well, which can make use of the preconfigured configuration obj instead of creating its own? What do u think?

Reply:
That was a typo fixed.

Regarding moving of methods, yes, I am thinking of doing some refactoring regarding listFiles but will do it after I write the tests.